onnx/version_converter/BaseConverter.h

```diff
@@ -0,0 +1,59 @@
+// Version converter interface for ONNX models between different opset versions.
+
+#pragma once
+
+#include "onnx/common/ir.h"
+#include "onnx/common/ir_pb_converter.h"
+#include "onnx/common/stl_backports.h"
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+#include <utility>
+#include <iostream>
+// TODO: Remove this import once actual adapters are imported
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+struct BaseVersionConverter {
+  // Schema for adapters: {<op_name>:{<from_domain>$<from_version>:{<to_domain>
+  // <to_version>: adapter}}}
+  std::unordered_map<std::string, std::unordered_map<std::string, std::unordered_map<std::string, Adapter*>>> adapters;
+
+  std::unordered_map<Node*, OpSchema*> current_opschemas;
+
+  BaseVersionConverter() {}
+
+  ~BaseVersionConverter() {
```

Comment:
Is it best practice to make constructors virtual, as well, so as to prevent variables instantiated as BaseConverter from solely invoking BaseConverter's methods?

Reply:
Constructors can't be virtual. Virtual means that it's deciding which function to call based on what the object actually is - but for a constructor call you don't have an object yet to base that decision on.